pkg/lib/authn/identity/ldap/provider.go

```diff
@@ -3,17 +3,26 @@ package ldap
 import (
 	"sort"
 
+	"github.com/authgear/authgear-server/pkg/api/model"
 	"github.com/authgear/authgear-server/pkg/lib/authn/identity"
 	"github.com/authgear/authgear-server/pkg/lib/authn/identity/service"
+	"github.com/authgear/authgear-server/pkg/lib/authn/stdattrs"
+	"github.com/authgear/authgear-server/pkg/lib/config"
+	"github.com/authgear/authgear-server/pkg/lib/ldap"
 	"github.com/authgear/authgear-server/pkg/util/clock"
 	"github.com/authgear/authgear-server/pkg/util/uuid"
 )
 
 var _ service.LDAPIdentityProvider = &Provider{}
 
+type StandardAttributesNormalizer interface {
+	Normalize(stdattrs.T) error
+}
```

Comment:
We should not define `StandardAttributesNormalizer` interface here **again**.

Reply:
Actually it is recommended to define it here again.

Local interfaces allows gomock to generate the mock, which make us able to write unit tests.